arwen/host/execution.go

```diff
@@ -117,29 +119,39 @@ func (host *vmHost) doRunSmartContractCall(input *vmcommon.ContractCallInput) (v
 	}()
 
 	result, returnCode, err := host.callSCMethod()
+
+	metering.UnlockGasIfAsyncStep()
+
 	if err != nil {
 		return vmOutput
```

Comment:
i think here returnCode should be set to something is err is not nil.

Reply:
just to make sure returnCode will yield error

Reply:
If `err != nil`, then `host.callSCMethod()` will have returned an appropriate value for `returnCode`. Then `err` is tested in the `defer` at the beginning of this method, along with the `returnCode`. Indeed, the code does not make this clear. Suggestions for rewriting?